apache-rat-core/src/main/java/org/apache/rat/OptionCollection.java

```diff
@@ -0,0 +1,638 @@
+            .desc(format("Specifies the level of detail in ARCHIVE file reporting. (default is %s)",
+                    ReportConfiguration.Processing.NOTIFICATION))
+            .converter(s -> ReportConfiguration.Processing.valueOf(s.toUpperCase()))
+            .build();
+
+    /**
+     * Specify the processing of STANDARD files.
+     */
+    public static final Option STANDARD = Option.builder().longOpt("standard").hasArg().argName("ProcessingType")
+            .desc(format("Specifies the level of detail in STANDARD file reporting. (default is %s)",
+                    Defaults.STANDARD_PROCESSING))
+            .converter(s -> ReportConfiguration.Processing.valueOf(s.toUpperCase()))
+            .build();
+
+    /**
+     * Ths option to signal the end of an argument list and the start of the directory/archive arguments.
+     */
+    // TODO rework when commons-cli 1.7.1 or higher is available.
+    static final DeprecatedAttributes DIR_ATTRIBUTES = DeprecatedAttributes.builder().setForRemoval(true).setSince("0.17")
+            .setDescription("Use '--'").get();
+    public static final Option DIR = Option.builder().option("d").longOpt("dir").hasArg()
+            .desc(format("[%s] %s", DIR_ATTRIBUTES, "Used to indicate end of list when using --exclude.")).argName("DirOrArchive")
+            .deprecated(DIR_ATTRIBUTES).build();
+
+    /**
+     * Option to signal that license text should be added to the files.
+     */
+    // TODO rework when Commons-CLI version 1.7.1 or higher is available.
+    private static final DeprecatedAttributes ADD_ATTRIBUTES = DeprecatedAttributes.builder().setForRemoval(true).setSince("0.17")
+            .setDescription("Use '-A' or '--addLicense' instead.").get();
+    static final OptionGroup ADD = new OptionGroup()
+            .addOption(Option.builder("a").hasArg(false)
+                    .desc(format("[%s]", ADD_ATTRIBUTES))
+                    .deprecated(ADD_ATTRIBUTES)
+                    .build())
+            .addOption(ADD_LICENSE);
+
+    /**
+     * Defines the copyright header to add to the file.
+     */
+    public static final Option COPYRIGHT = Option.builder().option("c").longOpt("copyright").hasArg()
+            .desc(format("The copyright message to use in the license headers, usually in the form of \"Copyright 2008 Foo\".  Only valid with --%s",
+                    ADD.getOptions().stream().filter(o -> !o.isDeprecated()).findAny().get().getLongOpt()))
+            .build();
+
+    /**
+     * Forces changes to be written to new files.
+     */
+    public static final Option FORCE = new Option("f", "force", false,
+            format("Forces any changes in files to be written directly to the source files (i.e. new files are not created).  Only valid with --%s",
+                    ADD.getOptions().stream().filter(o -> !o.isDeprecated()).findAny().get().getLongOpt()));
+
+
+    /*
+                    END OF OPTION LIST
```

Comment:
When I first scrolled the list of options I thought that it might make sense to add these option lists into an enum, that could provide parse methods or an if(RatOptionEnum) to do the parsing ..... not sure if this would help, but it would allow to encapsulate all of the options.
In later refactoring steps it could add export-methods that would convert it to Ant, Maven, CLI whatsoever configurations&options. WDYT?